pkg/parse/root.go

```diff
@@ -261,6 +261,18 @@ func setSourceStatusFields(source *v1beta1.SourceStatus, newStatus *SourceStatus
 	source.LastUpdate = newStatus.LastUpdate
 }
 
+func (p *root) setPreSyncAnnotations(ctx context.Context, commit string) status.Error {
+	rs := &v1beta1.RootSync{}
+	if err := p.Client.Get(ctx, rootsync.ObjectKey(p.SyncName), rs); err != nil {
+		return status.APIServerError(err, "failed to get RootSync for parser")
+	}
+	existing := rs.DeepCopy()
+	core.SetAnnotation(rs, metadata.SourceCommitAnnotationKey, commit)
+	core.SetAnnotation(rs, metadata.ImageURLAnnotationKey, p.Options.SourceRepo)
+	err := p.Client.Patch(ctx, rs, client.MergeFrom(existing), client.FieldOwner(configsync.FieldManager))
+	return status.SourceError.Wrap(err).Build()
```

Comment:
Suggest returning the patch error here and wrap it as a source error where it is called.

The intent is to wrap both the patch error and the get error in line 267 as a source error for consistency.

Reply:
Done